mitodl/mitxpro
# fix: configure course or program format

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxpro/issues/2611

#### What's this PR do?
Before this change, the format was hard-coded as `Online` for all courses and programs. In this PR, I made adjustments to the models so that the `Format` field can now be configured. Additionally, I set the default value of this field to `Online`. This PR also updates the serializer class used in the courses (`/api/courses`) and programs (`/api/programs`) APIs to allow for dynamic values for the `format` field. It also includes changes to the `seed_data` command so that it can now add the `format` value when creating courses and programs.

#### How should this be manually tested?
- First, run the necessary migrations to apply the changes locally.
- Next, make adjustments to the `Format` value within the Wagtail CMS for a course or program.
- Once you've made this change, navigate to the catalog page.
- Click on one of the products, and you should see the `Format` block displayed.

#### Screenshots (if appropriate)
<img width="955" alt="Screenshot 2023-09-12 at 4 08 34 PM" src="https://github.com/mitodl/mitxpro/assets/32649010/4733c2a7-00a0-42c9-b4ce-3d54935218a7">
